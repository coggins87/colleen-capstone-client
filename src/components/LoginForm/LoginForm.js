import React from 'react'
import AuthApiService from '../../services/auth-api-service.js'

class LoginForm extends React.Component{
  static defaultProps = {
    onLoginSuccess: ()=>{}
  }

  state = {error: null}

  handleSubmitJwtAuth = e => {
    e.preventDefault()
    this.setState({error: null})
    const {user_name, password} = e.target

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })
    .then(res=> {
      user_name.value=""
      password.value=""
      this.props.onLoginSuccess()
    })
    .catch(res=> {
      this.setState({error: res.error})
    })
    console.log('logged in')
  }
  render(){
    const {error}= this.state
    return (
      <form onSubmit={this.handleSubmitJwtAuth}>
        <div>{error &&<p>{error}</p>}</div>
      <label htmlFor='Login_user_name'>User Name:</label>
      <input required type='text' name='user_name' id='Login_user_name'></input>
      <label htmlFor='Login_password'>Password:</label>
      <input required type='password' name='password' id='Login_password'></input>
      <button type='submit'>Log In</button>
      </form>

    )
  }
}

export default LoginForm