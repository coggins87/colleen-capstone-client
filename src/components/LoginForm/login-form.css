._form {
  display: flex;
  flex-direction: column;
  width: 50vw;
  font-size: 1.2em;
}



.form_input {
  padding: 7px;
  font-size: 100%;
  margin: 2%;
}

.form_input:hover {
  border: 1px solid rgb(0, 183, 255);
}

.form_button {
  padding: 3%;
  margin-top: 2%;
  border: none;
  font-size: 1em;
  background-color: rgb(0, 183, 255);
  color: white;
}

.form_button:hover{
  background-color: rgb(0, 144, 201);
  color: white;
}

.form_input::placeholder{
  color: rgb(240, 18, 18);
  font-size: 75%;
}

.error {
  font-size: 2em;
}