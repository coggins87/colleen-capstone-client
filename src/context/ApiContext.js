import React from "react";
import TokenService from "../services/token-service";
import WorkoutService from "../services/workout-search-service";
const ApiContext = React.createContext({
  error: null,
  isLoggedIn: false,
  userId: null,
  setError: () => {},
  clearError: () => {},
  handleLoginSuccess: () => {}
});

export default ApiContext;

export class ApiProvider extends React.Component {
  state = {
    error: null,
    isLoggedIn: false,
    userId: null
  };
  setError = error => {
    console.error(error);
    this.setState({ error });
  };
  clearError = () => {
    this.setState({ error: null });
  };

  onDelete = () => {
    const user_id = this.props.reqs.params.userId;
    WorkoutService.deleteWorkout(user_id);
    this.setState({ workouts: WorkoutService.getUserWorkouts(user_id) });
  };

  nullError = () => {
    this.setState( {error: 'Invalid user, close this window to log out and try again'})
  }

  handleLoginSuccess = () => {
    const read = TokenService.readJwtToken();
    const user_Id = read.user_id;
    this.setState({
      userId: user_Id,
      isLoggedIn: true
    });
  };

  handleLogoutSuccess = () => {
    this.setState({
      userId: null,
      isLoggedIn: false
    });
  };

  isUserLoggedIn = () => {
    if (TokenService.hasAuthToken()) {
      const read = TokenService.readJwtToken();
      const user_Id = read.user_id;
      this.setState({
        userId: user_Id
      });
    }
  };
  render() {
    const value = {
      isUserLoggedIn: this.isUserLoggedIn,
      nullError: this.nullError,
      error: this.state.error,
      isLoggedIn: this.state.isLoggedIn,
      setError: this.setError,
      clearError: this.clearError,
      userId: this.state.userId,
      handleLoginSuccess: this.handleLoginSuccess,
      handleLogoutSuccess: this.handleLogoutSuccess,
      handleDeleteWorkout: this.onDelete
    };
    return (
      <ApiContext.Provider value={value}>
        {this.props.children}
      </ApiContext.Provider>
    );
  }
}
