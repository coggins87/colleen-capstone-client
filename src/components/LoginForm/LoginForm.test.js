import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginForm from "./LoginForm";
import { ApiProvider } from "../../context/ApiContext";

describe(`LoginForm component`, () => {
  const props = {
    testCheck: () => {}
  };

  it("renders a LoginForm by default", () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the LoginForm given props", () => {
    const wrapper = shallow(<LoginForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the LoginForm given context", () => {
    const wrapper = shallow(
      <ApiProvider>
        <LoginForm />
      </ApiProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
