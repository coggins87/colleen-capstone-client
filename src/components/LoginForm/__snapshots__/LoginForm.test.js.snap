// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a LoginForm by default 1`] = `
<form
  className="_form"
  onSubmit={[Function]}
>
  <div
    className="error"
  />
  <label
    htmlFor="Login_user_name"
  >
    User Name:
  </label>
  <input
    className="form_input"
    id="Login_user_name"
    name="user_name"
    placeholder="Required"
    required={true}
    type="text"
  />
  <label
    htmlFor="Login_password"
  >
    Password:
  </label>
  <input
    className="form_input"
    id="Login_password"
    name="password"
    placeholder="Required"
    required={true}
    type="password"
  />
  <button
    className="form_button"
    id="login_button"
    type="submit"
  >
    Log In
  </button>
</form>
`;

exports[`LoginForm component renders the LoginForm given context 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "handleDeleteWorkout": [Function],
      "handleLoginSuccess": [Function],
      "handleLogoutSuccess": [Function],
      "isLoggedIn": false,
      "isUserLoggedIn": [Function],
      "nullError": [Function],
      "setError": [Function],
      "userId": null,
    }
  }
>
  <LoginForm
    onLoginRedirect={[Function]}
  />
</ContextProvider>
`;

exports[`LoginForm component renders the LoginForm given props 1`] = `
<form
  className="_form"
  onSubmit={[Function]}
>
  <div
    className="error"
  />
  <label
    htmlFor="Login_user_name"
  >
    User Name:
  </label>
  <input
    className="form_input"
    id="Login_user_name"
    name="user_name"
    placeholder="Required"
    required={true}
    type="text"
  />
  <label
    htmlFor="Login_password"
  >
    Password:
  </label>
  <input
    className="form_input"
    id="Login_password"
    name="password"
    placeholder="Required"
    required={true}
    type="password"
  />
  <button
    className="form_button"
    id="login_button"
    type="submit"
  >
    Log In
  </button>
</form>
`;
